src/main/java/org/myrobotlab/programab/handlers/sraix/MrlSraixHandler.java

```diff
@@ -0,0 +1,58 @@
+package org.myrobotlab.programab.handlers.sraix;
+
+import java.util.Locale;
+
+import org.alicebot.ab.Chat;
+import org.alicebot.ab.SraixHandler;
+import org.myrobotlab.logging.LoggerFactory;
+import org.myrobotlab.programab.XmlParser;
+import org.myrobotlab.programab.handlers.oob.OobProcessor;
+import org.myrobotlab.programab.models.Sraix;
+import org.myrobotlab.service.ProgramAB;
+import org.myrobotlab.service.data.SearchResults;
+import org.myrobotlab.service.interfaces.SearchPublisher;
+import org.slf4j.Logger;
+
+public class MrlSraixHandler implements SraixHandler {
+  transient public final static Logger log = LoggerFactory.getLogger(MrlSraixHandler.class);
+
+  private ProgramAB programab = null;
+
+  public MrlSraixHandler(ProgramAB programab) {
+    this.programab = programab;
+  }
+
+  @Override
+  public String sraix(Chat chatSession, String input, String defaultResponse, String hint, String host, String botid, String apiKey, String limit, Locale locale) {
+    try {
+      log.debug("MRL Sraix handler! Input {}", input);
+      String xml = String.format("<sraix>%s</sraix>", input);

```

Comment:
Not sure this is a good way to handle this, any input containing non escaped XML characters are going to cause it to blow up

Reply:
its valid xml, its not a validated schema, but its valid xml

Reply:
Maybe do a try-catch, if it fails to parse then instead parse to an XML document (don't know what Jackson calls it, other libraries I've used let you deserialize to an internal type safe representation), then there should be a way to wrap it in the proper tags using type safe method calls

Reply:
There's a try/catch around the whole block, the the user should be informed of the problem code.
At this point I don't want to try to recover, informing the user of the issue hopefully will get them to correct it.